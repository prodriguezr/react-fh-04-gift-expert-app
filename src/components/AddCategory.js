import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handlerOnChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handlerOnSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length === 0) return;

    setCategories((oldCategories) => [inputValue, ...oldCategories]);

    setInputValue('');
  };

  return (
    <form onSubmit={handlerOnSubmit}>
      <input type='text' value={inputValue} onChange={handlerOnChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
