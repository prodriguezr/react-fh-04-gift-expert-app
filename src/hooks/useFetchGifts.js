import { useState, useEffect } from 'react';
import { getGifts } from '../helpers';

export const useFetchGifts = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifts(category).then((data) => {
      setState({
        data,
        loading: false,
      });
    });
  }, [category]);

  return state;
};
