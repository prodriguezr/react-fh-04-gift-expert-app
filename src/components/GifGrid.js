import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GifGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
