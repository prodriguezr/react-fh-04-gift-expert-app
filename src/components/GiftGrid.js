import { GiftGridItem } from './GiftGridItem';

import { useFetchGifts } from '../hooks';

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GiftGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};
