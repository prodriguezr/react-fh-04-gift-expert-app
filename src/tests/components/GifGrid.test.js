import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks';

jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid/>', () => {
  const category = 'One Punch';

  //   test('The category must be displayed within the h3 tag', () => {
  //     const wrapper = shallow(<GifGrid category={category} />);
  //     const h3 = wrapper.find('h3').text().trim();

  //     expect(h3).toBe(category);
  //   });

  //   test('It should display correctly', () => {
  //     const wrapper = shallow(<GifGrid category={category} />);

  //     expect(wrapper).toMatchSnapshot();
  //   });

  test('It should show items when images are loaded', () => {
    const data = [
      {
        id: 'dggg54Jh',
        url: 'https://localhost/images/87.png',
        title: 'Anything',
      },
    ];

    useFetchGifs.mockReturnValue({
      data,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(data.length);
  });
});
