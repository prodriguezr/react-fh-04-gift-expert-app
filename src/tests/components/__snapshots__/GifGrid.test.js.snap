// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid/> It should show items when images are loaded 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      key="dggg54Jh"
      title="Anything"
      url="https://localhost/images/87.png"
    />
  </div>
</Fragment>
`;
